src/physics/exact_solutions/exact_solution.cpp

```diff
@@ -0,0 +1,95 @@
+#include <deal.II/base/function.h>
+#include "exact_solution.h"
+
+namespace PHiLiP {
+
+// ========================================================
+// ZERO -- Returns zero everywhere; used a placeholder when no exact solution is defined.
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_Zero<dim,nstate,real>
+::ExactSolutionFunction_Zero()
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = 0;//time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_Zero<dim,nstate,real>
+::value(const dealii::Point<dim,real> &/*point*/, const unsigned int /*istate*/) const
+{
+    //NEED TO UPDATE THIS
+    //(void) point;
+    real value = 0;
+    return value;
+}
+
+// ========================================================
+// 1D SINE -- Exact solution for advection_explicit_time_study
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_1DSine<dim,nstate,real>
+::ExactSolutionFunction_1DSine (double time_compare)
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_1DSine<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) const
+{
+    double x_adv_speed = 1.0;
+
+    real value = 0;
+    real pi = dealii::numbers::PI;
+    if(point[0] >= 0.0 && point[0] <= 2.0){
+        value = sin(2*pi*(point[0] - x_adv_speed * t)/2.0);
+    }
+    return value;
+}
+
+
+//=========================================================
+// FLOW SOLVER -- Exact Solution Base Class + Factory
+//=========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction<dim,nstate,real>
+::ExactSolutionFunction ()
+    : dealii::Function<dim,real>(nstate)//,0.0) // 0.0 denotes initial time (t=0)
+{
+    //time_compare +=0;
+}
+
+template <int dim, int nstate, typename real>
+std::shared_ptr<ExactSolutionFunction<dim, nstate, real>>
+ExactSolutionFactory<dim,nstate, real>::create_ExactSolutionFunction(
+    Parameters::AllParameters const *const param)
+{
+
+    //read the final time
+    const double time_compare = param->flow_solver_param.final_time;
+
+    // Get the flow case type
+    const FlowCaseEnum flow_type = param->flow_solver_param.flow_case_type;
+    if (flow_type == FlowCaseEnum::advection_explicit_time_study) {
+        if constexpr (dim==1 && nstate==dim)  return std::make_shared<ExactSolutionFunction_1DSine<dim,nstate,real> > (time_compare);
+    } else {
+        // Select zero function if there is no exact solution defined
+        return std::make_shared<ExactSolutionFunction_Zero<dim,nstate,real>> ();
+    }
+    (void) time_compare; //to avoid 
```

Comment:
Delete unused line since it's being used now.

Reply:
Adjusted things so I don't need this line even if the zero function is selected.